src/main/java/com/google/sps/data/Video.java

```diff
@@ -70,15 +70,15 @@ public String getUrl(){
     public static EmbeddedEntity toEmbeddedEntity(Video video) {
         EmbeddedEntity videoEntity = new EmbeddedEntity();
         videoEntity.setProperty(URL_PROPERTY, video.getUrl());
-        videoEntity.setProperty(CURRENT_STATE_PROPERTY, video.getCurrentState().getValue());
+        videoEntity.setProperty(CURRENT_STATE_PROPERTY, (int) video.getCurrentState().getValue());
         videoEntity.setProperty(TIMESTAMP_PROPERTY, video.getCurrentTimeStamp());
         return videoEntity;
     }
 
     //Turns an embedded entity into a video object
     public static Video fromEmbeddedEntity(EmbeddedEntity videoEntity) {
         Map<String, Object> properties = videoEntity.getProperties();
-        return new Video((String) properties.get(URL_PROPERTY), VideoState.fromInt((int) properties.get(CURRENT_STATE_PROPERTY)), (long) properties.get(TIMESTAMP_PROPERTY));
+        return new Video((String) properties.get(URL_PROPERTY), VideoState.fromInt(((Long) properties.get(CURRENT_STATE_PROPERTY)).intValue()), (long) properties.get(TIMESTAMP_PROPERTY));
```

Comment:
Long vs long? Is it supposed to be the wrapper class?

Reply:
yup its a wrapper class